src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue

```diff
@@ -569,34 +639,46 @@ $spacing: 4px;
 	// keep inner border width in mind
 	$border-radius-outer: calc($border-radius + 2px);
 
-	&--button-variant &__label {
-		display: flex;
-		flex-direction: column;
-		justify-content: center;
-		width: 100%;
-		margin: 0;
+	&--button-variant.checkbox-radio-switch {
+		border: 2px solid var(--color-border-dark);
+		overflow: hidden;
+
+		&--checked {
+			font-weight: bold;
+
+			label {
+				background-color: var(--color-primary-light);
+			}
+		}
 	}
-	&--button-variant:not(&--button-variant-h-grouped) &__label {
-		align-items: flex-start;
+
+	// Text overflow of button style
+	&--button-variant &__label-text {
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+		width: 100%;
 	}
 
+	// Set icon color for non active elements to main text color
 	&--button-variant:not(&--checked) &__icon > * {
 		color: var(--color-main-text);
 	}
-	&--button-variant &__icon {
-		flex-basis: 100%;
 
-		// Hide icon container if empty to remove virtual padding
-		&:empty {
-			display: none;
-		}
+	// Hide icon container if empty to remove virtual padding
+	&--button-variant &__icon:empty {
+		display: none;
 	}
 
 	&--button-variant:not(&--button-variant-v-grouped):not(&--button-variant-h-grouped),
 	&--button-variant &__label {
 		border-radius: $border-radius;
 	}
 
+	/* Special rules for vertical button groups */
+	&--button-variant-v-grouped &__label {
+		flex-basis: 100%;
```

Comment:
```suggestion
		flex-basis: 100%;
		max-width: unset;
```
Should fix the issue mentioned above https://github.com/nextcloud/nextcloud-vue/pull/4065/files#r1191020846.